src/scripts/genDummyProvider.py

```diff
@@ -0,0 +1,231 @@
+##
+## Licensed to the .NET Foundation under one or more agreements.
```

Comment:
In general, I'm OK with the pattern that we generate all of these header and implementation files and then compile them into the product.  However, going forward, we should look at ways to simplify these files.  It looks to me like there are 3 copies that are almost identical (genDummyProvider.py, genEventPipe.py, and genEtwProvider.py).  If we are going to have to maintain all 3, we should look to make one implementation and override the functionality based on the file that we're generating.

Reply:
I definitely agree. These python scripts could use a good refactoring, the biggest gain of which could be code deduplication and a more class/module oriented design

Reply:
Great.  Do you have example header/implementation files that these generate for each platform that you can share?  I'd like to just confirm my understanding of what I'm seeing in code.

Reply:
For sure. Here is an example output https://git.io/vFYSq

Reply:
Thanks @nategraf.  Can you also share a sample FireEtw* function body as well?

Reply:
Here is an example: https://git.io/vFBSg

Reply:
Perfect.  Exactly what I was looking for.  Also, keep an eye out on https://github.com/dotnet/coreclr/pull/14706 - it's going to affect your change.